src/geo/transform.js

```diff
@@ -2059,19 +2065,38 @@ class Transform {
     // latitude and the center's latitude as you zoom in, camera to center distance varies dynamically.
     // As the cameraToCenterDistance is a function of zoom, we need to approximate the true zoom
     // given a mercator meter value in order to eliminate the zoom/cameraToCenterDistance dependency.
-    zoomFromMercatorZAdjusted(z: number): number {
-        const getZoom = (zoom) => {
-            const d = this.getCameraToCenterDistance(this.projection, zoom);
-            return this.scaleZoom(d / (z * this.tileSize));
+    zoomFromMercatorZAdjusted(mercatorZ: number): number {
+        assert(this.projection.name === 'globe');
+
+        const zoomFromMercatorZ = (zoom, mercatorZ) => {
+            assert(mercatorZ !== 0);
+            const worldSize = this.tileSize * Math.pow(2, zoom);
+            const d = this.getCameraToCenterDistance(this.projection, zoom, worldSize);
+            return this.scaleZoom(d / (mercatorZ * this.tileSize));
```

Comment:
Nit: this function is only used in one place so there's no need to make a closure, should be fine inline.